flocker/node/_docker.py

```diff
@@ -508,6 +508,9 @@ def _remove():
                     break
 
             try:
+                while (self._client.inspect_container(container_name)
+                        ['State']['Running']):
+                    sleep(0.1)
```

Comment:
On further investigation, I noticed two more things:

  * Many builders failed with a number of seemingly new Docker errors when building this branch.  It could be a coincidence or perhaps this change introduced that problem somehow (I can't really see how, but the point of this issue is that Docker interactions are subtle and hard to understand, I think...).  This probably merits further investigation.
  * It seems that containers with a restart policy interact poorly with the *stop* API.  If the container happens to not be running when the API call is processed, it fails with a 404.  Then Docker proceeds to restart the container according to its policy.  Our Docker layer interprets the 404 as meaning "it is safe to proceed to the *remove* step".  This is incorrect given the interaction with restart policies.  We probably need to do something different in the `NOT_FOUND` case in `_remove` in `DockerClient.remove`.